dot/state/block.go

```diff
@@ -360,9 +361,6 @@ func (bs *BlockState) HasBlockBody(hash common.Hash) (bool, error) {
 
 // GetBlockBody will return Body for a given hash
 func (bs *BlockState) GetBlockBody(hash common.Hash) (*types.Body, error) {
-	bs.RLock()
-	defer bs.RUnlock()
```

Comment:
Are we sure `GetBlockBody` is not used anywhere else? As in, maybe we need to add locking on other callers using this one?

Reply:
it isn't used anywhere else in the `BlockState`, so I think it's probably ok for now. but in the future it might be nice to do some lock overhaul on the `BlockState` since it could be improved imo.

Reply:
Maybe unexport that method to make this clearer, if possible?

Reply:
ah sorry it is used in other spots in the codebase, so gotta leave it like this for now

Reply:
So are you sure it won't data race on the block state if it's accessed elsewhere in the codebase?

Reply:
currently, the `GetBlockBody` just access the `unfinalisedBlocks` field that is a `*sync.Map` so by default can prevent concurrent access to the data other than that I'm pretty sure we can remove the locks

Reply:
yeah there's an edge case potentially with the `handleFinalisedBlock` function (just pushed a fix for it) previously that function was doing `LoadAndDelete` on the unfinalized blocks, then storing it in the db. so there was a potential race condition where one process was finalizing a block, and it was deleted from memory but not stored in the db yet, while another process calls  `GetBlockBody` and isn't able to find it in memory or in the db. however my update to `handleFinalisedBlock` should fix this (it now gets from memory, stores in the db, then deletes from memory)